cypress/integration/updateUser.spec.js

```diff
@@ -0,0 +1,731 @@
+/// <reference types="Cypress" />
+
+describe('Update User Testing', function () {
+  let info = {}
+  let randomInfo, profile, email
+  const newUsername = 'testuser6'
+  const newPassword = 'validpassword'
+
+  beforeEach(() => {
+    info = Cypress.env()
+    cy.visit('./cypress/integration/index.html').then((win) => {
+      expect(win).to.have.property('userbase')
+      cy.clearLocalStorage()
+    })
+
+    profile = {
+      a: 'a',
+      b: 'b',
+      c: 'c',
+    }
+
+    email = 'legit.email@example.com'
+
+    cy.getRandomInfoWithParams(email, profile, 'local').then((loginInfo) => {
+      randomInfo = loginInfo
+    })
+  })
+
+  after(() => {
+    cy.window().then((window) => {
+      const { userbase } = window
+      window._userbaseEndpoint = info.endpoint
+      userbase.init({ appId: info.appId })
+
+      return userbase.signIn({ username: newUsername, password: newPassword }).then(() => {
+        return userbase.deleteUser().then(() => {
+          console.log('Cleaning up updated user')
+          window.localStorage.clear()
+        })
+      }).catch(() => { })
+    })
+  })
+
+  it('Update user\'s e-mail, password and profile', function () {
```

Comment:
One additional note: can you add 1 additional test that updates everything **except** the username? The server logic is different when a username is or is not updated